packages/sanes-chrome-extension/src/logic/network/signerandprofile.ts

```diff
@@ -0,0 +1,33 @@
+import { MultiChainSigner, UserProfile } from '@iov/core';
+
+import { createUserProfile } from '../user';
+import { chainConnector } from './runtimeconfiguration';
+import { getConfigurationFile, codecTypeFromString } from '../config';
+import { singleton } from '../../utils/singleton';
+
+export interface SignerAndProfile {
+  readonly signer: MultiChainSigner;
+  readonly profile: UserProfile;
+}
+
+async function createSignerAndProfile(): Promise<SignerAndProfile> {
```

Comment:
One class returning two things, well... I also do not like it. I believe a more correct way to do it, as I suggested before, is moving the signer creation into Persona constructor. We break less "rules"

Reply:
Now everthing is created together in Persona.create(), so createSignerAndProfile was removed